### PR TITLE
fixes #254 replace google maps by leafletjs ans OSM

### LINKED ISSUE #254: Broken javascript on one GK page
The GeoKret GK6690 has it's page broken, from what I saw it is caused by the map Javascript.
The move located in Japan at coordinates `35.70210 139.77835` breaks all when used with other waypoints. The highest value I could have get for the longitude was `91`.

@boly38 Should we try to migrate to other library like leafletJS (#87) or upgrade to Google Map API v3 ? Would you like to work on this bug?

See: https://staging.geokrety.org/konkret.php?id=26256#

The loaded map file:
https://gist.github.com/kumy/2da4fb6c8b0e619e355d4e752a9b1c72

### PR DESCRIPTION
**Pull request by boly38**
fixes #254 replace google maps by leafletjs ans OSM
- leafletjs map container
- openstreetmap as map data
- TripRepository, CacheService
- TripService, Converters, ValidationService
- unit tests
- rest API rest/konkret/trip/read (doc into api.php)
- rebuild GPX and CSV exports
- phpcs
- update konkret and konkret-mapka

---
done tasks (for memory):
- [x] create php classes + unit test for this rest endpoint : <b>tripRepository</b>
- [x] tripRepository : add join with waypoint to avoid queries of queries..
- [x] tripRepository->getByGeokretyId : add geokrety not found exception
- [x] add CacheService : service that cache object in a file
- [x] add TripService : service that rely on tripRepository->getByGeokretyId  and cacheService
- [x] test : set and clean test dedicated cache directory
- [x] generate json that embed all results (leafletjs, csv, gpx)
- [x] add 'nocache' attribute to clean cache
- [x] move duplicated check methods into ValidationService class
- [x] leafletjs CDN 1.4.0
- [x] CSV : check cache with big comment > legacy cut the comment to 100 characters
- [x] keep csv gzip feature
- [x] csv link to update
- [x] GPX : comment cache comments to keep a light gpx file
- [x] gpx link to update
- [x] use config to define cache directory
- [x] map popup position (move to the bottom)
- [x] help + labs menus is masked by map
- [x] move rest api example to legacy doc (`api.php`)
- (postponed) query pagination by using offset : to do when replacing "konkret-tabelka"
- (postponed) new issue with ld json author

todo:
- [x] crlf
- [ ] test on staging

### CODE REVIEW

**Comment by kumy** [comment]
Will try it in a moment :)

**Comment by kumy** [comment]
Do you think this page could be also converted ?
* https://geokrety.house.kumy.net/mapka_kretow.php?userid=26422
* https://geokrety.house.kumy.net/user_stat.php?userid=26422

**Comment by kumy** [comment]
Just edited the `konkret.js` file to add an automatic zoom.

**Comment by kumy** [comment]
I really love this change. :clap: :crown:.

@boly38 Feel free to merge #269 then merge current PR to master.

File: website/lib/geokrety/service/CacheService.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        file_put_contents($cacheFile, $cacheContent);
+    }
+
+    public function evict($cacheId) {
+        $cacheFile = $this->getCacheFilename($cacheId);
+        file_exists($cacheFile) && unlink($cacheFile);
+    }
+
+    private function assumeCacheDirectory() {
+        file_exists($this->cacheDirectory) || mkdir($this->cacheDirectory, 0777, true);
```
[the remark this answers is not in the document]
Comment (boly38) [reply website/lib/geokrety/service/CacheService.php]: oops :) good catch

File: website/rest/konkret/trip/read.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            $csvFilename = 'trip_'.$geokretyId.'.csv';
+            $tripService->renderCsv($geokretyId, $trips, $csvFilename);
+            break;
+        default:
+            throw new Exception("format '$format' not supported, expected one of $supportedFormats");
+    }
+} catch (\Geokrety\Exception\NoDataException $noDataException) {
+    // handle errors
+    $errorResponse = new \Geokrety\Domain\ErrorResponse($action, $noDataException);
+    $errorResponse->write($format, 404);
```
[the remark this answers is not in the document]
Comment (boly38) [reply website/rest/konkret/trip/read.php]: I hesitate with an empty array
 but as i don't check if geokrety exists then
 I finally choose 'no trip found' == 404

[the remark this answers is not in the document]
Comment (kumy) [reply website/rest/konkret/trip/read.php]: Let's keep it as is, we can change things later.

**Comment by boly38** [comment]
other maps todo yes plus this one:
/mapki/google_static_logs.png

**Comment by boly38** [comment]
Codacy told me +61, but when I go to new issue I get 0 ???????
https://app.codacy.com/app/GeoKrety/geokrety-website/pullRequest?prid=3362592#newIssuesView

File: website/lib/geokrety/service/CacheService.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (boly38) [reply website/lib/geokrety/service/CacheService.php]: use 660 now

[the remark this answers is not in the document]
Comment (kumy) [reply website/lib/geokrety/service/CacheService.php]: Not sure `0660` is the right choice for a directory. It's equivalent to `rw-rw----`, processes will not be able to enter directory to list content.

[the remark this answers is not in the document]
Comment (boly38) [reply website/lib/geokrety/service/CacheService.php]: u're right, 
dumb I am ;)
will fix this next availability window

**Comment by boly38** [comment]
hi @kumy if you think we could keep 4 commits, then you could merge, else tell me , I will squash it
I wont take more time on codacy ;)

**Comment by kumy** [comment]
@boly38 don't hesitate to merge yourself to master :)

**Pull request merged**